code/menuui/techmenu.cpp

```diff
@@ -1223,64 +1183,39 @@ void techroom_init()
 
 void techroom_lists_reset()
 {
-	int i;
-
 	//unload the current animation, we load another one for the new current entry
-	if(Tab != SHIPS_DATA_TAB)
+	if (Tab != SHIPS_DATA_TAB)
 		techroom_unload_animation();
 
-	Current_list = NULL;
-	Current_list_size = 0;
-
 	model_free_all();
 	Techroom_ship_modelnum = -1;
 	Techroom_ship_model_instance = -1;
 
-	if (Ship_list != NULL) {
-		delete[] Ship_list;
-		Ship_list = NULL;
-	}
+	// This can be cleared immediately because there are no anims or bitmaps associated.
+	Ship_list.clear();
 
-	Ship_list_size = 0;
 	Ships_loaded = false;
 
-	if (Weapon_list != NULL) {
-		for (i = 0; i < Weapon_list_size; i++) {
-			if (Weapon_list[i].animation.num_frames != 0) {
-				generic_anim_unload(&Weapon_list[i].animation);
-			}
-
-			if (Weapon_list[i].bitmap >= 0) {
-				bm_release(Weapon_list[i].bitmap);
-				Weapon_list[i].bitmap = -1;
-			}
+	for (auto & list_item : Weapon_list) {
```

Comment:
This block is wrong and shouldn't be here. The animations have already been unloaded, and the rest of the contents in this loop should be *outside* of it. `model_free_all()` and clearing of the ships list is done twice here.